### ISSUE TITLE
WorkspaceEdit API didn't delete the created folder when undo. 

### ISSUE THREAD

**Issue by qweasd1** [microsoft/vscode#172463]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.74.3 
- OS Version: MacOS 12.3.1

Hi, 
When I develop my vscode extension with WorkspaceEdit API, I find the following issue:

Steps to Reproduce:

1. using WorkspaceEdit to create a new file inside a new folder. 
```typescript
const edit = new WorkspaceEdit();
edit.createFile(Uri.file(path.join("<a new folder that not exists>", 'someFile.tsx')));
await workspace.applyEdit(edit);
```
2. After ```await workspace.applyEdit(edit);``` is called, the new file and new folder is created
3. However, when I try to "Undo" inside the "someFile.tsx", I can see "someFile.tsx" is deleted, while the new folder is still there. Since the new folder is created by WorkspaceEdit, I thought it should also be removed.

**Comment by jrieken** [comment]
@bpasero Is there a way that `IWorkingCopyService#create` returns what folder it has created? I am calling it [here](https://github.com/microsoft/vscode/blob/16577239fe2a51f05a0387eedaab9ef566b32a3a/src/vs/workbench/contrib/bulkEdit/browser/bulkFileEdits.ts#L198) but I don't know what folders have been created by. My fallback would be to check for non-existing folders upfront

**Comment by bpasero** [comment]
@jrieken no, we do not even carry that information out of the file service:

https://github.com/microsoft/vscode/blob/65d0130a5c09ad99a0b35e1d3cb67deac8f6190f/src/vs/platform/files/common/fileService.ts#L386-L388
